src/core/link.h

```diff
@@ -5,9 +5,9 @@ enum { LINK_RING_SIZE    = 256,
 struct link {
   // this is a circular ring buffer, as described at:
   //   http://en.wikipedia.org/wiki/Circular_buffer
-  struct packet* packets[LINK_RING_SIZE];
+  struct packet *packets[LINK_RING_SIZE];
   struct {
-    double txbytes, rxbytes, txpackets, rxpackets, txdrop;
+    struct counter *txbytes, *rxbytes, *txpackets, *rxpackets, *txdrop;
```

Comment:
This change highlights a subtle issue with `shm` shared memory objects: you have to be very careful about storing pointers in them.

Generally speaking pointers are only valid in one process because each process has a private virtual address space. This looks like a bug on the `struct counter *` pointers: they will only be valid for the process that initializes the link and if another process uses those pointers it will lead to memory corruption or a SEGFAULT.

The exception is pointers that are valid in all processes. The `struct packet *` pointers are uniquely okay because we do actually ensure that the same packet has the same address in every process.

So a new solution has to be found for the counters: we can have them inline in the struct, or we can remove them entirely, but we can't have them as pointers (even to other shm objects because `shm.map()` does not return consistent addresses for a given object).

Aside: I am also having a really hard time tracking down a bug that is caused by setting these `struct counter *` pointers to counters allocated with `ffi.new()` instead of `shm.map()` but I think that this is unrelated.